### ISSUE TITLE
how to do offline install eta programming enviroment

### ISSUE THREAD

**Issue by clojurians-org** [typelead/eta#596]
i want to use eta on my work.
my personal PC has network connection while my work PC has no internet because of data security.
i can download whole eta staff on my personal PC, then send thoses file to my work PC.
but the question is, which files is needed by eta for offline work, or how to build a offline standalone package for developing?

**Comment by rahulmutt** [comment]
Are you allowed to build from source on your work PC? And also, what is the OS of your personal PC vs work PC? 

The process for building an offline package is as follows:
1. Install Etlas on your personal PC.
2. Run `etlas update`
3. Create a stub etlas project and add as the dependencies all the packages you want to use in your work computer.
4. Run `etlas install --dependencies-only` to make sure you fetch all the source tarballs that you require.
5. Copy the the following folders to the remote computer:
- `~/.etlas/packages/*`
- `~/.etlas/patches/*`

**Comment by clojurians-org** [comment]
thanks. i can build source code . i'll try it.
my personal PC OS is MacOS, the work PC OS is REDHAT 6.
and i has the private maven repository that will proxy for downloading the maven package.

**Comment by chengenzhao** [comment]
We got the same problem, mac also
when executing eta update, it stuck there
and we tried to use gradle plugin but it seems like it is not in maven central repository?

**Comment by rahulmutt** [comment]
@chengenzhao Yes, `etlas` maintains a separate repository. The function of the Gradle plugin is to notify gradle of the locations of the Eta-compiled jar files. We do eventually plan to start uploading packages to Maven but we are currently working out the binary compatibility story.

**Comment by jneira** [comment]
I think you'll maybe have to comment [this line](https://github.com/typelead/eta/blob/master/shake/Build.hs#L205) of eta-build to get a succesfull build
